pkg/network/usm/ebpf_gotls.go

```diff
@@ -233,6 +233,10 @@ func (p *GoTLSProgram) ConfigureOptions(options *manager.Options) {
 		options.MapEditors = make(map[string]*ebpf.Map)
 	}
 
+	options.MapSpecEditors[probes.SockByPidFDMap] = manager.MapSpecEditor{
+		MaxEntries: p.cfg.MaxTrackedConnections,
+		EditorFlag: manager.EditMaxEntries,
+	}
```

Comment:
why? we're getting the map FD and link it in the next line

Reply:
We switched internally to using a non-deprecated way of doing the map editing, and it enforces that the key size, value size, map type, and *max entries* match before doing the replacement.

Reply:
and if it does not match, then we have a failure?

Reply:
Yes, correct.